tests/foreman/ui_airgun/test_activationkey.py

```diff
@@ -66,6 +67,79 @@ def module_org():
     return entities.Organization().create()
 
 
+@tier2
+@upgrade
+def test_positive_end_to_end_crud(session, module_org):
+    """Perform end to end testing for activation key component
+
+    :id: b6b98c45-e41e-4c7a-9be4-997273b7e24d
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    name = gen_string('alpha')
+    new_name = gen_string('alpha')
+    cv = entities.ContentView(organization=module_org).create()
+    cv.publish()
+    with session:
+        session.activationkey.create({
+            'name': name,
+            'lce': {ENVIRONMENT: True},
+            'content_view': cv.name,
+        })
+        assert session.activationkey.search(name)[0]['Name'] == name
+        ak_values = session.activationkey.read(name)
+        assert ak_values['details']['name'] == name
+        assert ak_values['details']['content_view'] == cv.name
+        # Update architecture with new name
+        session.activationkey.update(name, {'details.name': new_name})
+        assert session.activationkey.search(new_name)[0]['Name'] == new_name
+        assert not session.activationkey.search(name)
+        # Delete architecture
+        session.activationkey.delete(new_name)
+        assert not session.activationkey.search(new_name)
+
+
+@tier3
+@upgrade
+def test_positive_end_to_end_register(session):
+    """Create activation key and use it during content host registering
+
+    :id: dfaecf6a-ba61-47e1-87c5-f8966a319b41
+
+    :expectedresults: Content host was registered successfully using activation
+        key, association is reflected on webUI
+
+    :CaseLevel: System
+
+    :CaseImportance: High
+    """
+    org = entities.Organization().create()
+    lce = entities.LifecycleEnvironment(organization=org).create()
+    repos_collection = RepositoryCollection(
+        distro=DISTRO_RHEL7,
+        repositories=[SatelliteToolsRepository()]
+    )
+    repos_collection.setup_content(org.id, lce.id, upload_manifest=True)
+    ak_name = repos_collection.setup_content_data['activation_key']['name']
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+        repos_collection.setup_virtual_machine(vm)
+        with session:
+            session.organization.select(org.name)
+            chost = session.contenthost.read(vm.hostname)
+            assert (
+                    chost['details']['registered_by'] == 'Activation Key {}'
+                    .format(ak_name)
+            )
+            ak_values = session.activationkey.read(ak_name)
+            assert len(ak_values['content_hosts']['table'])
+            assert ak_values[
```

Comment:
is it one liner for a new rule?
Of course, it is a nitpick, but I think that rule was introduced exactly for such lines

not a blocker